tasks/internal/create-advisory-task/create-advisory-task.yaml

```diff
@@ -221,15 +227,30 @@ spec:
               echo "Remaining entries after filtering:"
               cat "$CONTENT_FILE"
 
+              CONTENT_BEFORE_COUNT=$(jq 'length' <<< "$CONTENT_BEFORE_FILTER")
+              CONTENT_AFTER_COUNT=$(jq 'length' "$CONTENT_FILE")
+              if [[ $CONTENT_BEFORE_COUNT -gt $CONTENT_AFTER_COUNT ]]; then
+                if [[ -z "$LATEST_ADVISORY_FILE" ]]; then
+                  LATEST_ADVISORY_FILE="$ADVISORY_FILE"
+                  FILTERED_COUNT=$((CONTENT_BEFORE_COUNT - CONTENT_AFTER_COUNT))
+                  echo "Tracked latest advisory: $LATEST_ADVISORY_FILE (filtered $FILTERED_COUNT items)"
+                fi
+              fi
+
               # If after filtering, no entries are left, then we can exit early
               if jq -e 'length == 0' "$CONTENT_FILE" >/dev/null; then
-                  echo "Matching advisory exists: ${ADVISORY_FILE}. Skipping creation."
-                  ADVISORY_INTERNAL_URL="${GIT_REPO//\.git/}/-/raw/main/${ADVISORY_FILE}"
-                  ADVISORY_TYPE=$(yq -r '.spec.type' "${ADVISORY_FILE}")
-                  ADVISORY_NAME=$(yq -r '.metadata.name' "${ADVISORY_FILE}")
+                  echo "All content found in existing advisories. Skipping creation."
+                  
+                  FINAL_ADVISORY_FILE="${LATEST_ADVISORY_FILE:-$ADVISORY_FILE}"
```

Comment:
We set it when an image is filtered out, so why do we need the default? Is there any scenario where it would get this far and not be set?